### ISSUE TITLE
Leviathan trade execution is severely suboptimal

### ISSUE THREAD

**Issue by masonwheeler** [cameroncondry/cbc-kitten-scientists#333]
Due to the way trading affects their departure schedule, the optimal strategy for trading with Leviathans is to wait to make the first trade until you max out Unobtainium.  The script does not take this into account, eagerly making the first trade as soon as possible, thereby squandering a great deal of trading potential.

The rule should be, `if ((the leviathans' days remaining > the point it will be reduced to after the first trade) && (unobtainium quantity < full)) do not trade;`

**Comment by rherban** [comment]
I agree that this is suboptimal and should be updated, but it can be mitigated by unchecking the `limited` flag.  

But there is a slightly-related issue where (IMO) it should trade just before the leviathans disappear, regardless of how much you have saved up.
